docs/tileentities/tileentity.md

```diff
@@ -0,0 +1,128 @@
+# TileEntities
+
+Tile Entities are like simplified Entities, that are bound to a Block.
+They are used to store dynamic data, execute tick based tasks and for dynamic rendering.
+Some examples would be, vanilla Minecraft to handle inventories on chests, smelting logic on furnaces, or area effects for beacons. 
+More advanced examples exist in mods, such as quarries, sorting machines, pipes, and displays.
+
+!!! note
+    `TileEntities` aren't a solution for everything and they can cause lag when used in large amounts.
+    When possible, try to avoid them.
+
+## Creating a `TileEntity`
+In order to create a `TileEntity` you need to extend the `TileEntity` class.
+It is important that your `TileEntity` has a default constructor, so that Minecraft can properly load it.
+After you've created your class, you need to register the `TileEntity`. For this you need to call 
+```JAVA
+    GameRegistry#registerTileEntity(Class<? extends TileEntity> tileEntityClass, ResourceLocation key)
+```
+Where the first parameter is simply your TileEntity class and the second the registry name of your TileEntity.
+
+!!! note
+    The method to register a `TileEntity` is using a `String` instead of a `ResourceLocation` before Forge version `14.23.3.2694`
+    This method creates a `ResourceLocation` from the String. Be sure to use the ResourceLocation format modid:tile_entity to avoid it being called minecraft:tile_entity.
+
+## Attaching a `TileEntity` to a `Block`
+To attach your new `TileEntity` to a `Block` you need to override 2 (two) methods within the Block class.
+```JAVA
+    Block#hasTileEntity(IBlockstate state)
+
+    Block#createTileEntity(World world, IBlockState state)
+```
+Using the parameters you can choose if the block should have a `TileEntity` or not.
+Usually you will return `true` in the first method and a new instance of your `TileEntity` in the second method.
+
+At this point the `TileEntity`can be used for Rendering.
+
+## Storing Data within your `TileEntity`
+In order to save data, override the following two methods
+```JAVA
+    TileEntity#writeToNBT(NBTTagCompound nbt)
+
+    TileEntity#readFromNBT(NBTTagCompound nbt)
+```
+These methods are called whenever the `TileEntity` gets (un-)loaded.
```

Comment:
more correct to say it's called whenever the chunk is loaded from, or saved to, NBT.